plugins/nvme/fu-nvme-device.c

```diff
@@ -286,6 +287,11 @@ fu_nvme_device_parse_cns (FuNvmeDevice *self, const guint8 *buf, gsize sz, GErro
 	s1ro = buf[260] & 0x01;
 	g_debug ("fawr: %u, nr fw slots: %u, slot1 r/o: %u", fawr, nfws, s1ro);
 
+	/* FRU globally unique identifier (FGUID) */
+	gu = fu_nvme_device_get_guid_safe (buf, 127);
+	if (gu != NULL)
+		fu_device_add_guid (FU_DEVICE (self), gu);
```

Comment:
Looking through https://nvmexpress.org/wp-content/uploads/NVM_Express_Revision_1.3.pdf and https://nvmexpress.org/wp-content/uploads/NVM_Express_Management_Interface_1_0_gold.pdf it looks to me that this is new to NVME 1.3, correct?

I'm starting to wonder if there needs to be a decoder for NVME GUIDs?  Or at least a howto interpret which GUID to use for directing updates?

Reply:
Yes, this is new to 1.3 -- I only noticed it when I was looking at the new FWUG stuff. I can certainly add this to the README.md

Reply:
Yes I think add to the `README` is important and also mention in the readme the way to launch `fwupdtool` with correct `--daemon-verbose` options to see which GUIDs are being used for which strings during development.

Reply:
I've added it to the README -- and I also think we're making it way too hard to find the actual DeviceInstanceID's for the GUIDs. I don't think we should be asking people to use fwupdtool or run the daemon in verbose mode just to query something so they can write a metainfo file. Maybe we should just add something like `DeviceInstanceIDs` as a property on each of the FuDevices? We could then show them in `fwupdmgr get-devices` or maybe just in `fwupdmgr get-devices --verbose`